dbms/src/Functions/FunctionsStringSearch.h

```diff
@@ -126,6 +236,17 @@ class FunctionsStringSearch : public IFunction
         const auto * col_haystack_const = typeid_cast<const ColumnConst *>(&*column_haystack);
         const auto * col_needle_const = typeid_cast<const ColumnConst *>(&*column_needle);
 
+        MutableColumnPtr ilike_col0_copy_ptr;
```

Comment:
I think there is a standard way to do this:
```
void executeImpl(Block & result_block, const ColumnNumbers & arguments, size_t result) const override
{
        auto block = result_block;
        if constexpr (name == std::string_view(NameIlike3Args::name)) 
        {
                block.getByPosition(arguments[0]).column = (*std::move(result_block.getByPosition(arguments[0]).column)).mutate();
                block.getByPosition(arguments[1]).column = (*std::move(result_block.getByPosition(arguments[1]).column)).mutate(); 
                lowerEnglishWords(block, arguments);
        }
        /// original code
        /// ....
        /// before return
        result_block.getByPosition(result).column = std::move(block.getByPosition(result).column);
}
```

Reply:
> I think there is a standard way to do this:
> 
> ```
> void executeImpl(Block & result_block, const ColumnNumbers & arguments, size_t result) const override
> {
>         auto block = result_block;
>         if constexpr (name == std::string_view(NameIlike3Args::name)) 
>         {
>                 block.getByPosition(arguments[0]).column = (*std::move(result_block.getByPosition(arguments[0]).column)).mutate();
>                 block.getByPosition(arguments[1]).column = (*std::move(result_block.getByPosition(arguments[1]).column)).mutate(); 
>                 lowerEnglishWords(block, arguments);
>         }
>         /// original code
>         /// ....
>         /// before return
>         result_block.getByPosition(result).column = std::move(block.getByPosition(result).column);
> }
> ```

okk